ee/packages/license/src/MockedLicenseBuilder.ts

```diff
@@ -190,13 +197,17 @@ export class MockedLicenseBuilder {
 		return this;
 	}
 
-	public withGratedModules(modules: LicenseModule[]): this {
+	public withGratedModules(modules: string[]): this {
 		this.grantedModules = this.grantedModules ?? [];
-		this.grantedModules.push(...modules.map((module) => ({ module })));
+		this.grantedModules.push(
+			...(modules.map((module) =>
+				CoreModules.includes(module as LicenseModule) ? { module } : { module, external: true },
+			) as GrantedModules),
+		);
 		return this;
 	}
 
-	withNoGratedModules(modules: LicenseModule[]): this {
+	withNoGratedModules(modules: string[]): this {
```

Comment:
Why string removing specificity here?

Reply:
Please check the ADR above for details on the decision for add-ons (external modules)

Reply:
You can make a string alias like `type ExternalModule = string`, or better yet:
```ts
type ExternalModule = `${string}.${string}`;
```

to limit the format of extermal modules to strings with at least one period (the domain separator);
Then you can make `type LicenseModule = CoreModule | ExternalModule`.

Reply:
You can also use this to associate `external: false` with `InternalModule` and `external: true` with `ExternalModule`, which would remove the need of doing additional typeguards on the code below.

Reply:
I'm checking internally on the naming convention (I've used it in the tests but it is not confirmed)

Reply:
Alright, naming convention applies :)